xmtp/src/mock_xmtp_api_client.rs

```diff
@@ -53,7 +53,8 @@ impl Clone for MockXmtpApiClient {
     }
 }
 
-#[async_trait]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
```

Comment:
Out of curiosity, what is the reasoning behind this? We need `Send` on non-wasm, but we don't need it on wasm?

Reply:
Yep, that's right -- I know, it's an eyesore :( ([more info here](https://github.com/rustwasm/wasm-bindgen/issues/2833)). 

More generally, I don't love the look of this yet. But I was thinking to revisit it after / as I sort out how to implement the `subscribe()` call since I suspect it will have the gnarlier concurrency issues.